File: valid_anagram/Python/anagram.py
# 242. Valid Anagram
# Definition: An anagram is a word, phrase, or name formed by rearranging the
#             letters of another, such as cinema, formed from iceman.

def Anagram_Hash(s: str, t: str) -> bool:
    if len(s) != len(t):
        return False

    countS, countT = {}, {} 

    for i in range(len(s)):
        countS[s[i]] = countS.get(s[i], 0) + 1
        countT[t[i]] = countT.get(t[i], 0) + 1

    for c in countS:
        if countS[c] != countT.get(c, 0):
            return False

    return True


def Anagram_Sort(s: str, t: str) -> bool:
    return sorted(s) == sorted(t)

File: valid_anagram/JS/anagram.js
// 242. Valid Anagram

const Anagram_Hash = (s, t) => {
  if (s.length != t.length) return false;

  let sArray = s.split("").sort();
  let tArray = t.split("").sort();

  let i = 0;

  while (i < s.length) {
    if (sArray[i] !== tArray[i]) return false;
    i++;
  }
  return true;
};

const Anagram_Sort = (s, t) => {
  let sArray = s
    .split("")
    .sort((a, b) => (a > b ? 1 : -1))
    .join();
  let tArray = t
    .split("")
    .sort((a, b) => (a > b ? 1 : -1))
    .join();

  return sArray === tArray;
};

module.exports = { Anagram_Hash, Anagram_Sort };
